wsd/DocumentBroker.cpp

```diff
@@ -471,8 +471,10 @@ void DocumentBroker::pollThread()
             _admin.addBytes(getDocKey(), deltaSent, deltaRecv);
         }
 
-        if (_storage && _lockCtx->needsRefresh(now))
+        if (_storage && !_lockStateUpdateRequest && _lockCtx->needsRefresh(now))
+        {
```

Comment:
Remark: So we need refreshLock() also if the _lockStateUpdateRequest is not yet set, i.e. desired if not required.

refreshLock() then retries to use a writable and authorized session and propagates down to (old: updateStorageLockState()) now updateStorageLockStateAsync().
Within this path I assume _lockStateUpdateRequest will be set eventually? 
Yes, updateStorageLockStateAsync() assigns the single _lockStateUpdateRequest entity.

Reply:
Additional thought: Feasible to have the _lockCtx hold _lockStateUpdateRequest, so it also could use it for the `needsRefresh()` reply?

Reply:
I think the `_lockCtx` should be part of the Storage instance. 

What this is doing is simply checking that there isn't already an async request in flight. If there is, we don't need to issue a second one.

Perhaps we can simplify it, but I rather deal with `_lockCtx` more intelligently first.